### ISSUE TITLE
Logging out and logging in to a different account shows the old account

### ISSUE THREAD

**Issue by clonejo** [streetcomplete/StreetComplete#5574]
**How to Reproduce**
I wanted to switch to a different account, and after logging back in, when clicking on "my profile", the account i just logged out of gets shown. I think the star count shown in the map overlay is also from the old account.

In the login flow, after clicking on "authorize" i sometimes get a blank page.

I don't know if i am still properly authenticated to the old account, and i did not want to try by uploading for privacy reasons.

After a couple of logout+login attempts (many ending in that blank page) the correct account shows up.

**Expected Behavior**
All personal data related to the old account must be wiped on logout. Local settings should be kept. It should be possible to keep not uploaded answers, in case one forgot to log out before entering a couple answers.

**Versions affected**
57.1 from f-droid

**Comment by westnordost** [comment]
This is weird. I am pretty sure that on pressing the logout button, all statistics are cleared from the database, i.e. the star count should also be gone.

I recently (two weeks ago) refactored the whole screen. It should of course work the same, but it is impossible to say whether this bug persists with the new implementation (v57.1 was released four weeks ago). For reference, it should look like this:

https://github.com/streetcomplete/StreetComplete/assets/4661658/96e03c0e-38c9-4f82-b1d3-b7002dec54d8

It's not asking for my username and password because I checked the "remember me" button on the website. Also, the statistics are empty for a second or so because after downloading it is downloading the name, avatar and statistics in the background just after login.

Anyway, I will close this as (maybe) solved because the implementation changed pretty much completely.
